src/firebase/firebase_auth.cr

```diff
@@ -25,6 +25,38 @@ module Google
 
     @user_agent : String
 
+    ####################
+    # Users - accounts #
+    ####################
+
+    # API details: https://cloud.google.com/identity-platform/docs/reference/rest/v1/accounts/signUp
+    def sign_up_request(email : String? = nil, password : String? = nil, display_name : String? = nil, **opts)
```

Comment:
I have an inkling that there should be a helper that converts the stylistically correct snake-cased keys of `**opts` to google's camel-cased options.

Not extremely necessary but something about the potential for...

`client.sign_up_request(display_name: "yadayadah", hasBlueRibbon: true)` 

doesn't feel right.

But also it is also something that will need to be documented.

Reply:
Would be nice, but also quite complex to handle nested objects.

Reply:
The case it would need to handle is when **opts includes objects and lists
```
#sign_up_request(display_name: "yadayadah", mfaInfo: {displayName: "mobile", phoneInfo: "04 9999 9999"})
```

Reply:
A really rough implementation that could use fewer intermediate allocations for what's occurring.
```crystal
def transform_options(args : NamedTuple | Hash)
  hash = args.is_a?(Hash) ? args : args.to_h
  hash.transform_keys(&.to_s.underscore).transform_values do |value|
    if value.is_a?(NamedTuple) || value.is_a?(Hash)
      transform_options(value)
    else
      value
    end
  end
end
```

Reply:
@caspiano 
I have added opts converter method and it's applied to everything except for the `#lookup_request# method where it results in a type error.

```sh
In /home/mia/.asdf/installs/crystal/1.5.0/share/crystal/src/iterator.cr:864:29

 864 | Map(typeof(self), T, U).new(self, func)
                               ^--
Error: no overload matches 'Iterator::Map(Hash::EntryIterator(Symbol, NoReturn), Tuple(Symbol, NoReturn), Tuple(String, Array(String))).new' with types Hash::EntryIterator(Symbol, NoReturn), Proc(Tuple(Symbol, Array(String)), Tuple(String, Array(String)))
```